packages/salesforcedx-vscode-core/src/metadataSupport/metadataHoverProvider.ts

```diff
@@ -0,0 +1,255 @@
+      const [fullMatch, , elementName] = match;
+      const matchStart = match.index;
+      const matchEnd = match.index + fullMatch.length;
+
+      // Check if cursor is within this element
+      if (cursorPosition >= matchStart && cursorPosition <= matchEnd) {
+        // Remove namespace prefix if present
+        const cleanElementName = elementName.includes(':') ? elementName.split(':')[1] : elementName;
+
+        // Check if this looks like a metadata type (starts with capital letter)
+        if (/^[A-Z]/.test(cleanElementName)) {
+          return cleanElementName;
+        }
+      }
+    }
+
+    // Also check if the word itself is a metadata type
+    if (/^[A-Z]/.test(word) && word.length > 2) {
+      return word;
+    }
+
+    return null;
+  }
+
+  /**
+   * Extract field information for internal tags within metadata
+   */
+  private extractFieldInfo(
+    document: vscode.TextDocument,
+    position: vscode.Position
+  ): { metadataType: string; fieldName: string } | null {
+    const line = document.lineAt(position.line);
+    const wordRange = document.getWordRangeAtPosition(position);
+
+    if (!wordRange) {
+      return null;
+    }
+
+    // Look for XML element patterns for fields (lowercase or camelCase)
+    const xmlElementRegex = /<(\/?)([\w:]+)(\s|>|\/)/g;
+    let match;
+
+    while ((match = xmlElementRegex.exec(line.text)) !== null) {
+      const [fullMatch, , elementName] = match;
+      const matchStart = match.index;
+      const matchEnd = match.index + fullMatch.length;
+
+      // Check if cursor is within this element
+      if (position.character >= matchStart && position.character <= matchEnd) {
+        // Remove namespace prefix if present
+        const cleanElementName = elementName.includes(':') ? elementName.split(':')[1] : elementName;
+
+        // Check if this looks like a field (not a metadata type)
+        if (!/^[A-Z]/.test(cleanElementName) && cleanElementName.length > 1) {
+          // Find the parent metadata type by scanning upward
+          const parentType = this.findParentMetadataType(document, position.line);
+          if (parentType) {
+            return { metadataType: parentType, fieldName: cleanElementName };
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Find the parent metadata type by scanning upward from current line
+   */
+  private findParentMetadataType(document: vscode.TextDocument, startLine: number): string | null {
+    for (let i = startLine; i >= 0; i--) {
+      const line = document.lineAt(i).text;
+      const xmlElementRegex = /<([\w:]+)(\s|>)/g;
+      let match;
+
+      while ((match = xmlElementRegex.exec(line)) !== null) {
+        const elementName = match[1];
+        const cleanElementName = elementName.includes(':') ? elementName.split(':')[1] : elementName;
+
+        // Check if this looks like a metadata type (starts with capital letter)
+        if (/^[A-Z]/.test(cleanElementName)) {
```

Comment:
hmm, if we have the list of all valid metadata types (ex: from SDR registry), would that be useful vs. "any uppercase letter"?